### PR TITLE
Kills java instances for Azure pipelines

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: eng/scripts/KillProcesses.ps1 [hunk 1]
```diff
@@ -10,6 +10,12 @@ function _kill($processName) {
     }
 }
 
+function _killSeleniumJavaInstances(){
```
Comment (ryanbrandenburg) [inline eng/scripts/KillProcesses.ps1]: nit: This name isn't quite accurate, it's killing our java instances, which might be selenium. Don't block on it though.

**Review by ryanbrandenburg — APPROVED** [review]

File: eng/scripts/KillProcesses.ps1 [hunk 2]
```diff
@@ -22,6 +28,8 @@ _kill vctip.exe
 _kill chrome.exe
 _kill h2spec.exe
 _kill WerFault.exe
+_killSeleniumJavaInstances
```
Comment (dougbu) [inline eng/scripts/KillProcesses.ps1]: Why isn't this `_kill java.exe`? or a few `_kill` calls for the Java processes we use?

**Review by dougbu — CHANGES_REQUESTED** [review]
Update KillProcesses.sh too

File: eng/scripts/KillProcesses.ps1 (hunk 2 above)
Comment (dougbu) [reply eng/scripts/KillProcesses.ps1]: To be clear: I won't block on this but it looks more complicated that necessary and will fail the build if a `Stop-Process` fails (unlike in the `_kill` function).

Comment (javiercn) [reply eng/scripts/KillProcesses.ps1]: I already fixed that locally. I'm trying it out on a different PR. I will update this one and merge it once  I see it working on the other one.

File: eng/scripts/KillProcesses.ps1 [hunk 3]
```diff
@@ -10,6 +10,18 @@ function _kill($processName) {
...
 
+function _killJavaInstances() {
+    $_javaProcesses = Get-Process java -ErrorAction SilentlyContinue |
+        Where-Object { $_.Path -like "$env:JAVA_HOME*" };
+    foreach($_javaProcess in $_javaProcesses) {
+        try {
+            Stop-Process $proc
+        } catch {
+            Write-Host "Failed to kill java process: $proc 
+        }
```
Comment (dougbu) [inline eng/scripts/KillProcesses.ps1]: This looks more robust.

**Review by dougbu — APPROVED** [review]
I'm still curious: Why is `_kill java.exe` or a couple of similar commands insufficient? And, why isn't it useful to kill Java processes on non-Windows machines i.e. in KillProcesses.sh (that only kills `dotnet`)?

File: eng/scripts/KillProcesses.ps1 [hunk 4]
```diff
@@ -10,6 +10,18 @@ function _kill($processName) {
...
 }
 
+function _killJavaInstances() {
+    $_javaProcesses = Get-Process java -ErrorAction SilentlyContinue |
+        Where-Object { $_.Path -like "$env:JAVA_HOME*" };
+    foreach($_javaProcess in $_javaProcesses) {
+        try {
+            Stop-Process $proc
+        } catch {
+            Write-Host "Failed to kill java process: $proc
```
Comment (Eilon) [inline eng/scripts/KillProcesses.ps1]: Is there a missing end quote here?

Comment (javiercn) [reply eng/scripts/KillProcesses.ps1]: Good catch

**Comment by Eilon** [comment]
Force merged.

**Comment by Eilon** [comment]
Thanx @javiercn !

**Pull request merged**